lib/screens/explore_screens/regions_screens/regions_main/interactor/viewmodel/region_bloc.dart

```diff
@@ -13,21 +20,61 @@ part 'region_state.dart';
 
 class RegionBloc extends Bloc<RegionEvent, RegionState> {
   RegionBloc(RegionModel? region) : super(RegionState.initial(region)) {
+    on<OnRegionMounted>(_onRegionMounted);
     on<OnJoinRegionButtonPressed>(_onJoinRegionButtonPressed);
+    on<OnLeaveRegionButtonPressed>(_onLeaveRegionButtonPressed);
   }
 
   Stream<List<RegionMessageModel>> get regionMessages =>
       FirebaseDatabaseRegionsRepository().getMessagesForRegion(state.region!.id);
 
+  Future<void> _onRegionMounted(OnRegionMounted event, Emitter<RegionState> emit) async {
+    if (state.region == null) {
```

Comment:
should we move this logic to a state mapper?

Reply:
I have 2 uses cases here I'm not really sure how to hanndle both in a single mapper.
Thas is why only the las part has a mapper.